applications/materialDataCreator/appMaterialDataCreator.js

```diff
@@ -0,0 +1,280 @@
+
+    button.clicked.connect(clicked);
+    
+    function createDemo() {
+        demoID = Entities.addEntity({
+                    type: "Box",
+                    name: "MATERIAL DEMO",
+                    dimensions: {
+                        x: 1,
+                        y: 1,
+                        z: 1
+                    },
+                    grab: {
+                        grabbable: true
+                    },
+                    shape: "Cube",
+                    position: Vec3.sum(MyAvatar.position, Vec3.multiplyQbyV(MyAvatar.orientation, { x: 0, y: 0.25, z: -5 }))
+                },"local");
+        
+        
+        
+        materialID = Entities.addEntity({
+                    parentID: demoID,
+                    parentMaterialName: "0",
+                    type: "Material",
+                    name: "untitled",
+                    materialURL: "materialData",
+                    priority: 1,
+                    materialData: "{\"materials\":{}}",
+                    position: Vec3.sum(MyAvatar.position, Vec3.multiplyQbyV(MyAvatar.orientation, { x: 0, y: 0.25, z: -3 }))
+                },"local");
+    }
+
+    function updateDemo(matData) {
+        var data = JSON.parse(matData);
+        var materialName = data.materials[0].name;
+        Entities.editEntity(materialID, {
+                name: materialName,
+                materialData: matData
+            });
+    }
+
+    function deleteDemo() {
+        Entities.deleteEntity(materialID);
+        Entities.deleteEntity(demoID);
+    }
+
+    function onMoreAppWebEventReceived(message) {
+        eventObj = JSON.parse(message);
+        if ( eventObj.chanel === chanel) {
+            if ( eventObj.action === "updateMaterialData") {
+                currentMaterialRecord = eventObj.materialRecord;
+                currentMaterialData = eventObj.materialData;
+                updateDemo(eventObj.materialData);
+            }
+            if ( eventObj.action === "changeDemoShape") {
+                changeDemoShape(eventObj.shape);
+            }
+            if ( eventObj.action === "createMaterialEntity") {
+                var newMaterialEntityId = Entities.addEntity({
+                    parentMaterialName: "0",
+                    type: "Material",
+                    name: currentMaterialRecord.name,
+                    materialURL: "materialData",
+                    priority: 1,
+                    materialData: currentMaterialData,
+                    position: Vec3.sum(MyAvatar.position, Vec3.multiplyQbyV(MyAvatar.orientation, { x: 0, y: 0.25, z: -2 }))
+                },"domain");
+                if (newMaterialEntityId !== Uuid.NULL) {
+                    audioFeedback.confirmation();
+                }
+            }
+            if ( eventObj.action === "exportMaterialAsJson") {
+                Window.saveFileChanged.connect(onFileSaveChanged);
+                Window.saveAsync("Select Where to Save", "", "*.json");

```

Comment:
Grammatically this would be
```suggestion
                Window.saveAsync("Select where to save", "", "*.json");
```
but I don't know what this is so it might make sense anyways.